AutoCollection/ServerRequestParser.ts

```diff
@@ -89,13 +91,29 @@ class ServerRequestParser extends RequestParser {
         newTags[ServerRequestParser.keys.sessionId] = tags[ServerRequestParser.keys.sessionId] || this._getId("ai_session");
         newTags[ServerRequestParser.keys.userId] = tags[ServerRequestParser.keys.userId] || this._getId("ai_user");
         newTags[ServerRequestParser.keys.userAgent] = tags[ServerRequestParser.keys.userAgent] || this.userAgent;
-        newTags[ServerRequestParser.keys.operationName] = tags[ServerRequestParser.keys.operationName] || this.method + " " + url.parse(this.url).pathname;
-        newTags[ServerRequestParser.keys.operationParentId] = tags[ServerRequestParser.keys.operationParentId] || this.parentId;
-        newTags[ServerRequestParser.keys.operationId] = tags[ServerRequestParser.keys.operationId] || this.operationId;
+        newTags[ServerRequestParser.keys.operationName] = this.getOperationName(tags);
+        newTags[ServerRequestParser.keys.operationParentId] = this.getOperationParentId(tags);
+        newTags[ServerRequestParser.keys.operationId] = this.getOperationId(tags);
 
         return newTags;
     }
 
+    public getOperationId(tags:{[key: string]:string}) {
+        return tags[ServerRequestParser.keys.operationId] || this.operationId;
+    }
+
+    public getOperationParentId(tags:{[key: string]:string}) {
+        return tags[ServerRequestParser.keys.operationParentId] || this.parentId || this.getOperationId(tags);
+    }
+
+    public getOperationName(tags:{[key: string]:string}) {
+        return tags[ServerRequestParser.keys.operationName] || this.method + " " + url.parse(this.url).pathname;
```

Comment:
was this ` url.parse(this.url).pathname` already parsed out of the requestTelemetry? Maybe just use `requestData.name`?